src/vehicle_use.cpp

```diff
@@ -2212,19 +2212,21 @@ void vehicle::build_interact_menu( veh_menu &menu, const tripoint &p, bool with_
     const cata::optional<vpart_reference> vp_door_lock = vp.avail_part_with_feature( "DOOR_LOCKING" );
     if( vp_lockable_door && vp_door_lock && !vp_lockable_door->part().open ) {
         const size_t door_idx = vp_lockable_door->part_index();
-        if( !vp_lockable_door->part().locked ) {
+        if( player_inside && !vp_lockable_door->part().locked ) {
             menu.add( string_format( _( "Lock %s" ), vp_lockable_door->part().name() ) )
             .hotkey( "LOCK_DOOR" )
             .on_submit( [this, door_idx] {
                 this->lock_or_unlock( door_idx, true );
+                get_player_character().moves -= 100;
```

Comment:
Using `mod_moves` would be preferred over directly manipulating moves.

Reply:
Done.